### ISSUE TITLE
Support Basic Auth for "Raw" when repo is private.

### ISSUE THREAD

**Issue by frankhommers** [gitbucket/gitbucket#1535]
This is a feature request: I would like to have basic auth for requests to the raw url on private repo's. 

This way I could wget the data still without having git onto the target system.

**Comment by kounoike** [comment]
@frankhommers I added such feature in Web API: [ApiController.scala:633](https://github.com/gitbucket/gitbucket/blob/2d9727a6959bb53ad98a91b79c8fd12a97016241/src/main/scala/gitbucket/core/controller/ApiController.scala#L633). You can use basic auth and personal access token in Web API.

**Comment by frankhommers** [comment]
@kounoike Yeah, I've seen that. But that doesn't result directly in the raw file, but it's one step closer ;-)

**Comment by frankhommers** [comment]
@aadrian, how is this LDAP/SSO related? It pretty much just not that ;-)
